File: src/main/java/org/geepawhill/contentment/grid/Horizontal.java
package org.geepawhill.contentment.grid;

import org.geepawhill.contentment.geometry.PointPair;

public class Horizontal
{
	PointPair points;

	public Horizontal(PointPair bounds, double percent)
	{
		double y = bounds.along(percent / 100d).getY();
		points = new PointPair(bounds.getFrom().getX(),y, bounds.getTo().getX(),y);
	}

	public PointPair points()
	{
		return points;
	}

}

File: src/main/kotlin/org/geepawhill/contentment/grid/Horizontal.kt
package org.geepawhill.contentment.grid

import org.geepawhill.contentment.geometry.PointPair

class Horizontal(bounds: PointPair, percent: Double) {
    private var points: PointPair

    init {
        val y = bounds.along(percent / 100.0).y
        points = PointPair(bounds.from.x, y, bounds.to.x, y)
    }

    fun points(): PointPair {
        return points
    }

}
